### ISSUE TITLE
frontmatter date property not re-evaluated

### ISSUE THREAD

**Issue by meredrica** [zk-org/zk#478]
### Check if applicable

- [X] I have searched the existing issues (**required**)
- [ ] I'm willing to help fix the problem and contribute a pull request

### Describe the bug

If you create a note with a `date:` frontmatter property it is not honored/updated until you delete the `.zk/notebook.db` file

seems similar to #382 

### How to reproduce?

- create a new file `echo '# test' > foo.md`
- `zk list` correctly shows `(just now)`
- update the file `zk edit foo.md` with this content:
```yaml
---
date: 2020-01-01 00:00:00
---

# test
```
- run `zk list`, output is `(just now)`.
- run `zk index -f`, nothing changes
- `rm .zk/notebook.db`
- `zk list` date is correct

### zk configuration

```toml
default zk config after zk init.
```

### Environment

```bash
zk 0.14.1
system: Linux 6.11.9-arch1-1 x86_64 GNU/Linux
```

**Comment by tjex** [comment]
Yeah, this is a known issue and has to do with zk indexing the date / time on creation. So if you use a template that inserts the date and time on creation, then it will be referenced properly. It's not a bad idea to periodically delete and re-index the database anyway, so I'm not considering this as too urgent given I have limited bandwidth at the moment with starting a new job next week and maintainer activity here has effectively whittled down to me alone (which is fine, just painting more of the picture).
